ticdc/ticdc-architecture.md

```diff
@@ -0,0 +1,180 @@
+- processor CheckpointTS: Indicates the minimum table CheckpointTS on the Processor.
+- global CheckpointTS: Indicates the minimum CheckpointTS among all Processors .
+
+Generally, a checkpointTS satisfies the following constraint:
+
+```
+table CheckpointTS >= global CheckpointTS
+```
+
+Because TiCDC only replicates data smaller than the global ResolvedTS to the downstream, the complete constraint is as follows:
+
+```
+table ResolvedTS >= global ResolvedTS >= table CheckpointTS >= global CheckpointTS
+```
+
+After data changes and transactions are committed, the ResolvedTS on the TiKV node will continue to advance, and the Puller module on the TiCDC node keeps receiving data pushed by TiKV. The Puller module also decides whether to scan incremental data based on the data changes it has received, which ensures that all data changes are sent to the TiCDC node.
+
+The Sorter module sorts data received by the Puller module in ascending order according to the timestamp. This process ensures data consistency at the table level. Next, the Mounter module assembles the data changes from the upstream into a format that the Sink module can consume, and sends it to the Sink module. The Sink module replicates the data changes between the CheckpointTS and the ResolvedTS to the downstream in the order of the timestamp, and advances the checkpointTS after the downstream receives the data changes.
+
+The preceding sections only cover data changes of DML statements and do not include DDL statements. The following sections introduce the timestamp related to DDL statements.
+
+#### Barrier TS
+
+Barrier TS is generated when a DDL statement is executed or a Syncpoint is used.
+
+- This timestamp ensures that all changes before this DDL statement are replicated to the downstream. After this DDL statement is executed and replicated, TiCDC starts replicating other data changes. Because DDL statements are processed by the Capture Owner, the Barrier TS corresponding to a DDL statement is only generated by the Processor thread of the owner node.
+- Syncpoint Barrier TS is also a timestamp. When you enable the Syncpoint feature of TiCDC, a Barrier TS is generated by TiCDC according to the `sync-point-interval` you specified. When all table changes before this Barrier TS are replicated, TiCDC records the global Checkpoint downstream, from which data replication continues next time.
+
+After a Barrier TS is generated, TiCDC only replicates data changes that occur before this Barrier TS to downstream. Then TiCDC checks whether all target data has been replicated by comparing the global CheckpointTS and Barrier TS. If global CheckpointTS equals to Barrier TS, TiCDC continues replication after performing a designated operation (such as executing a DDL statement or recording the global CheckpointTS downstream). Otherwise, TiCDC waits for all data changes that occur before Barrier TS to be replicated to the downstream.
```

Comment:
```suggestion
After a Barrier TS is generated, TiCDC only replicates data changes that occur before this Barrier TS to downstream. Then TiCDC checks whether all target data has been replicated by comparing the global CheckpointTS and Barrier TS. If global CheckpointTS equals to Barrier TS, TiCDC continues replication after performing a designated operation (such as executing a DDL statement or recording the global CheckpointTS in downstream). Otherwise, TiCDC waits for all data changes that occur before Barrier TS to be replicated to the downstream.
```